Flakey FCGI test

jetty-10

See jenkins run:
https://jenkins.webtide.net/blue/organizations/jenkins/jetty.project/detail/PR-7153/3/pipeline

```
[ERROR] org.eclipse.jetty.fcgi.server.HttpClientTest.testLongPollIsAbortedWhenClientIsStopped  Time elapsed: 0.12 s  <<< FAILURE!
java.lang.AssertionError: 
Connection Leaks
Expected: is <0L>
     but: was <1L>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.eclipse.jetty.fcgi.server@10.0.8-SNAPSHOT/org.eclipse.jetty.fcgi.server.AbstractHttpClientServerTest.dispose(AbstractHttpClientServerTest.java:105)

```

--- a/jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java
+++ b/jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java
@@ -459,13 +459,29 @@ public abstract class AbstractConnectionPool extends ContainerLifeCycle implemen
     @Override
     public void close()
     {
-        // Manually release and remove entries to do our best effort calling the listeners.
-        for (Pool<Connection>.Entry entry : pool.values())
+        // Forcibly release and remove entries to do our best effort calling the listeners.
+        try
         {
-            if (entry.release())
-                released(entry.getPooled());
-            if (entry.remove())
-                removed(entry.getPooled());
+            for (Pool<Connection>.Entry entry : pool.values())
+            {
+                while (entry.isInUse())
+                {
+                    if (entry.release())
+                    {
+                        released(entry.getPooled());
+                        break;
+                    }
+                }
+                if (entry.remove())
+                    removed(entry.getPooled());
+            }
+        }
+        catch (Throwable x)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Detected concurrent modification while forcibly releasing the pooled connections", x);
+            // We could not call the listeners for all entries, but at least the following
+            // pool.close() call will still release all resources.
         }
         pool.close();
     }

--- a/jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java
+++ b/jetty-client/src/main/java/org/eclipse/jetty/client/AbstractConnectionPool.java
@@ -459,6 +459,14 @@ public abstract class AbstractConnectionPool extends ContainerLifeCycle implemen
     @Override
     public void close()
     {
+        // Manually release and remove entries to do our best effort calling the listeners.
+        for (Pool<Connection>.Entry entry : pool.values())
+        {
+            if (entry.release())
+                released(entry.getPooled());
+            if (entry.remove())
+                removed(entry.getPooled());
+        }
         pool.close();
     }
 

--- a/jetty-client/src/main/java/org/eclipse/jetty/client/LeakTrackingConnectionPool.java
+++ b/jetty-client/src/main/java/org/eclipse/jetty/client/LeakTrackingConnectionPool.java
@@ -42,8 +42,8 @@ public class LeakTrackingConnectionPool extends DuplexConnectionPool
     @Override
     public void close()
     {
-        LifeCycle.stop(this);
         super.close();
+        LifeCycle.stop(this);
     }
 
     @Override

